kyaml/yaml/types_test.go

```diff
@@ -166,3 +167,86 @@ type: string
 	}
 	assert.Equal(t, expected, actual)
 }
+
+func TestIsMissingOrNull(t *testing.T) {
+	if !IsMissingOrNull(nil) {
+		t.Fatalf("input: nil")
+	}
+	// missing value or null value
+	node := &RNode{value: nil}
+	if !IsMissingOrNull(node) {
+		t.Fatalf("input: nil value")
+	}
+
+	node.value = &yaml.Node{}
+	if IsMissingOrNull(node) {
+		t.Fatalf("input: valid node")
+	}
+	// node with NullNodeTag
+	node.value.Tag = NullNodeTag
+	if !IsMissingOrNull(node) {
+		t.Fatalf("input: with NullNodeTag")
+	}
+
+	node.value = &yaml.Node{}
+	if IsMissingOrNull(node) {
+		t.Fatalf("input: valid node")
+	}
+}
+
+func TestIsEmpty(t *testing.T) {
+	if !IsEmpty(nil) {
+		t.Fatalf("input: nil")
+	}
+
+	// missing value or null value
+	node := &RNode{value: nil}
+	if !IsEmpty(node) {
+		t.Fatalf("input: nil value")
+	}
+	// not array or map
+	node.value = &yaml.Node{}
+	if IsEmpty(node) {
+		t.Fatalf("input: not array or map")
+	}
+}
+
+func TestIsEmpty_Arrays(t *testing.T) {
+	node := &RNode{value: &yaml.Node{
+		Kind: yaml.SequenceNode,
+	}}
+	// empty array. empty array is not expected as empty
+	if IsEmpty(node) {
+		t.Fatalf("input: empty array")
+	}
+	// array with 1 item
+	node.value.Content = append(node.value.Content, &yaml.Node{})
+	if IsEmpty(node) {
+		t.Fatalf("input: array with 1 item")
+	}
+	// delete the item in array
+	node.value.Content = node.value.Content[:len(node.value.Content)-1]
+	if IsEmpty(node) {
+		t.Fatalf("input: empty array")
+	}
+}
+
+func TestIsEmpty_Maps(t *testing.T) {
+	node := &RNode{value: &yaml.Node{
+		Kind: yaml.MappingNode,
+	}}
```

Comment:
is there a public way to create a MappingNode ?  Would it make sense to have such a method?